lib/src/ruby/api.cc

```diff
@@ -485,21 +486,47 @@ namespace facter { namespace ruby {
         }
         LOG_DEBUG("ruby was found at \"%1%\".", ruby);
 
-        bool success;
-        string output, none;
-        tie(success, output, none) = execute(ruby, { "-e", "print File.join(RbConfig::CONFIG['"+libruby_configdir()+"'], RbConfig::CONFIG['LIBRUBY_SO'])" });
+        bool success, loaded = false;
+        string ruby_cmd_args, ruby_output, libruby_file, libruby_full_path, none;
+        tie(success, libruby_file, none) = execute(ruby, { "-e", "print RbConfig::CONFIG['LIBRUBY_SO']"});
```

Comment:
@peterhuene updated with a first pass of reducing ruby calls. I was able to query for all three libruby dirs in one go, but I had to do one other call to get `LIBRUBY_SO`. Previously we were executing File.join(RbConfig::CONFIG($dirname), RbConfig::CONFIG('LIBRUBY_SO')), but I ran into a situation where ruby returned nil for 'archlibdir' (on OSX), causing File.join to error out, even though there were other directories to check.

We could get `LIBRUBY_SO` in the same call as the three libruby directories, but we'd have to do some additional string parsing to save it and remove it from the list so we don't accidentally try and check an obviously invalid path. Does that sound like something we'd want to do here?